hubble-relay/cmd/root.go

```diff
@@ -40,9 +41,27 @@ func New() *cobra.Command {
 	}
 	logger := logging.DefaultLogger.WithField(logfields.LogSubsys, "hubble-relay")
 	vp := newViper()
+	flags := rootCmd.PersistentFlags()
+	flags.BoolP("debug", "D", false, "Enable debug messages")
+	vp.BindPFlags(flags)
+
+	// We need to check for the debug environment variable or CLI flag before
+	// loading the configuration file since on configuration file read failure
+	// we will emit a debug log entry.
+	if vp.GetBool("debug") {
+		logging.SetLogLevel(logrus.DebugLevel)
+	}
 	if err := vp.ReadInConfig(); err != nil {
-		logger.WithError(err).Warnf("Failed to read config from file '%s'", configFilePath)
+		logger.WithError(err).Debugf("Failed to read config from file '%s'", configFilePath)
```

Comment:
I would prefer at least `Info` level. Because in a real deployment done using our Helm charts, the config file should always be present. So if the config file is absent, this likely indicates a deployment problem and the user should be able to see this message without enabling debug.

Reply:
That's fair. The issue is that then we're back to square one where any `./hubble-relay version` or `hubble-relay completion bash` will display this log entry right?

Reply:
Ah, I did not realize that complication. I guess we could check if `vp.ConfigFileUsed()` existed in `runServe`, but that would still not catch all possible errors (such as permission problems). Let's keep this PR as is then!